tests/test_profile.py

```diff
@@ -14,3 +24,19 @@ def test_unique_cands():
     unique_cands = {"A", "B", "C", "E"}
     assert unique_cands == set(cands)
     assert len(cands) == len(unique_cands)
+
+
+def test_updates_in_place():
+    before = test_profile.get_ballots()
+    remove = "a"
+    after = remove_cand(remove, before)
+    assert remove in test_profile.get_candidates()
+    for ballot in after:
+        if remove in ballot.ranking:
```

Comment:
`if remove in ballot.ranking: assert remove in ballot.ranking` is never going to fail, worth revisiting to decide what you actually want to test here